src/txmempool.cpp

```diff
@@ -642,28 +654,22 @@ bool CTxMemPool::addUnchecked(const uint256 &hash,
     cachedInnerUsage += entry.DynamicMemoryUsage();
 
     const CTransaction &tx = newit->GetTx();
-    std::set<uint256> setParentTransactions;
     for (unsigned int i = 0; i < tx.vin.size(); i++)
     {
-        mapNextTx[tx.vin[i].prevout] = CInPoint(&tx, i);
-        setParentTransactions.insert(tx.vin[i].prevout.hash);
-    }
-    // Don't bother worrying about child transactions of this one.
-    // Normal case of a new transaction arriving is that there can't be any
-    // children, because such children would be orphans.
-    // An exception to that is if a transaction enters that used to be in a block.
-    // In that case, our disconnect block logic will call UpdateTransactionsFromBlock
-    // to clean up the mess we're leaving here.
+        mapNextTx.emplace(tx.vin[i].prevout, CInPoint{&tx, i});
 
-    // Update ancestors with information about this tx
-    for (const uint256 &phash : setParentTransactions)
-    {
-        txiter pit = mapTx.find(phash);
+        // Update ancestors with information about this tx
+        txiter pit = mapTx.find(tx.vin[i].prevout.hash);
```

Comment:
In the case where a tx spends N inputs from the same parent tx, the old code will execute ```_UpdateParent(newit, pit, true);``` 1 time, but the new code will execute it N times.

Reply:
yes you're right, i'll revert that part.